File: imperative-scripting-languages/practical_6.pl
#!/usr/bin/perl
use strict;
use warnings;

if (@ARGV != 2) {
    print "Usage: perl script.pl <filename> <word>\n";
    exit(1);
}

my ($filename, $word) = @ARGV;

if (-e $filename) {
    open my $fh, '<', $filename or die "Could not open '$filename' $!\n";
    my $fileContent = do { local $/; <$fh> };
    close $fh;

    $fileContent = lc $fileContent;
    $fileContent =~ s/[^a-z]//g;

    my $reversedContent = reverse $fileContent;

    my $count = () = $reversedContent =~ /$word/g;

    print "Matches: $count\n";
} else {
    print "File not found.\n";
}

File: imperative-scripting-languages/practical_6.php
<?php
    if ($argc != 3) {
        echo "Usage: php script.php <filename> <word>\n";
        exit(1);
    }

    $filename = $argv[1];
    $word = $argv[2];

    if (file_exists($filename)) {
        $fileContent = file_get_contents($filename);
        $fileContent = strtolower($fileContent);
        $fileContent = preg_replace("/[^a-z]/", "", $fileContent);

        $reversedContent = strrev($fileContent);

        $count = substr_count($reversedContent, $word);

        $htmlOutput = <<<HTML
            <!DOCTYPE html>
            <html>
            <head>
                <meta charset="UTF-8">
                <title>Word Count</title>
            </head>
            <body>
                <p>Converted String: $fileContent</p>
                <p>Reversed Word: $reversedContent</p>
                <p>Matches: $count</p>
            </body>
            </html>
            HTML;

        echo $htmlOutput;
    } else {
        $htmlOutput = <<<HTML
            <!DOCTYPE html>
            <html>
            <head>
                <meta charset="UTF-8">
                <title>Word Count</title>
            </head>
            <body>
                <p>File not found.</p>
            </body>
            </html>
            HTML;

        echo $htmlOutput;
    }
?>